internal/cli/login.go

```diff
@@ -82,15 +84,39 @@ func (cmd *loginCommand) Handler(profile *Profile, ui terminal.UI, args []string
 
 	profile.SetUser(cmd.publicAPIKey, cmd.privateAPIKey)
 
-	res, loginErr := cmd.realmClient.Login(cmd.publicAPIKey, cmd.privateAPIKey)
-	if loginErr != nil {
-		return loginErr
+	auth, authErr := cmd.realmClient.Authenticate(cmd.publicAPIKey, cmd.privateAPIKey)
+	if authErr != nil {
+		return authErr
 	}
 
-	profile.SetSession(res.AccessToken, res.RefreshToken)
+	profile.SetSession(auth.AccessToken, auth.RefreshToken)
 	return profile.Save()
 }
 
 func (cmd *loginCommand) Feedback(profile *Profile, ui terminal.UI) error {
 	return ui.Print(terminal.TextMessage{"Successfully logged in."})
 }
+
+const (
+	loginErrorMsg = "login failed"
+)
+
+type loginError struct {
+	reason string
+}
+
+func (err loginError) Error() string {
+	var sb strings.Builder
+
+	if _, err := sb.WriteString(loginErrorMsg); err != nil {
+		return loginErrorMsg
+	}
+
+	if err.reason != "" {
+		if _, err := sb.WriteString(fmt.Sprintf(": %s", err.reason)); err != nil {
+			return loginErrorMsg
+		}
+	}
```

Comment:
this seem overly complex - could this just become:

```
if err.reason != "" {
    return fmt.Sprintf("%s: %s", loginErrorMsg , err.reason)
}
return loginErrorMsg
```

Reply:
Ugh, this is my bad for creating such a mess of PRs here, but this code is/should be gone away.  instead I am wrapping any errors caught from the `Handler` interface in `cli/cli.go` like this:

```go
		RunE: func(c *cobra.Command, a []string) error {
			return cmdDef.Command.Handler(
			err := cmdDef.Command.Handler(
				factory.Profile,
				newCobraUI(factory.Config.UI, c),
				a,
			)

			if err != nil {
				return fmt.Errorf("%s failed: %w", display, err)
			}
			return nil
		},
```